data_store/controllers/retrieve_submission_file.py

```diff
@@ -71,10 +71,13 @@ def get_custom_file_name(submission_id: str) -> str:
         .one_or_none()
     )
 
-    date = submission_info.submission_date.strftime("%Y-%m-%d")
-    start_date = submission_info.observation_period_start.strftime("%b%Y")
-    end_date = submission_info.observation_period_end.strftime("%b%Y")
+    if submission_info:
+        date = submission_info.submission_date.strftime("%Y-%m-%d")
+        start_date = submission_info.observation_period_start.strftime("%b%Y")
+        end_date = submission_info.observation_period_end.strftime("%b%Y")
 
-    file_name = f"{date}-{submission_info.fund_code}-{submission_info.organisation_name}-{start_date}-{end_date}"
+        file_name = f"{date}-{submission_info.fund_code}-{submission_info.organisation_name}-{start_date}-{end_date}"
+    else:
+        file_name = f"data-return-{submission_id}"
```

Comment:
I think this fallback filename could be made nicer. The `submission_id` variable here is just a UUID, but the S3 metadata for the file should contain the human-readable code.

Unfortunately our naming here is not great - `submission.id` is a UUID, `submission.submission_id` is the friendly version. The latter should be available on the S3 object, and it also has the programme name.

So maybe in the fallback case we can build the filename from the S3 metadata. We can still get: the date of the submission (should be something like the object's CreatedAt value), the friendly submission ID and the LA name (metadata `programme_name`).

I realise this is a bit more work, but feels like the nice thing to do and the best information we can provide.

Thoughts?

Reply:
Updated the code to get user friendly filename.